apps/ledger-live-desktop/src/renderer/modals/EditTransaction/steps/StepSummary.jsx

```diff
@@ -0,0 +1,258 @@
+import styled from "styled-components";
+import {
+  getAccountCurrency,
+  getAccountName,
+  getAccountUnit,
+  getMainAccount,
+} from "@ledgerhq/live-common/account/index";
+import Box from "~/renderer/components/Box";
+import Button from "~/renderer/components/Button";
+import CryptoCurrencyIcon from "~/renderer/components/CryptoCurrencyIcon";
+import Ellipsis from "~/renderer/components/Ellipsis";
+import FormattedVal from "~/renderer/components/FormattedVal";
+import Text from "~/renderer/components/Text";
+import IconQrCode from "~/renderer/icons/QrCode";
+import IconWallet from "~/renderer/icons/Wallet";
+import { rgba } from "~/renderer/styles/helpers";
+import CounterValue from "~/renderer/components/CounterValue";
+import type { ThemedComponent } from "~/renderer/styles/StyleProvider";
+import type { StepProps } from "../types";
+import AccountTagDerivationMode from "~/renderer/components/AccountTagDerivationMode";
+import { BigNumber } from "bignumber.js";
+
+const FromToWrapper: ThemedComponent<{}> = styled.div``;
+const Circle: ThemedComponent<{}> = styled.div`
+  height: 32px;
+  width: 32px;
+  border-radius: 32px;
+  background-color: ${p => rgba(p.theme.colors.palette.primary.main, 0.1)};
+  color: ${p => p.theme.colors.palette.primary.main};
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  margin-right: 12px;
+`;
+const VerticalSeparator: ThemedComponent<{}> = styled.div`
+  height: 18px;
+  background: ${p => p.theme.colors.palette.text.shade20};
+  width: 1px;
+  margin: 1px 0px 0px 15px;
+`;
+const Separator: ThemedComponent<{}> = styled.div`
+  height: 1px;
+  background: ${p => p.theme.colors.palette.text.shade20};
+  width: 100%;
+  margin: 15px 0;
+`;
+
+export default class StepSummary extends PureComponent<StepProps> {
+  render() {
+    const { account, parentAccount, transaction, status, transactionRaw } = this.props;
+    if (!account) return null;
+    const mainAccount = getMainAccount(account, parentAccount);
+    if (!mainAccount || !transaction) return null;
+    const { estimatedFees, totalSpent } = status;
+    const amount = transaction.amount;
+    if (amount.isEqualTo(BigNumber(0))) {
+      // increase gas fees in case of cancel flow as we don't have the fees input screen for cancel flow
+      if (transactionRaw.maxFeePerGas) {
+        transaction.maxFeePerGas = BigNumber(
+          BigNumber(transactionRaw.maxFeePerGas)
+            .multipliedBy(1.3)
+            .toFixed(),
+        );
+      }
+      if (transactionRaw.maxPriorityFeePerGas) {
+        transaction.maxPriorityFeePerGas = BigNumber(
+          BigNumber(transactionRaw.maxPriorityFeePerGas)
+            .multipliedBy(1.1)
+            .toFixed(),
+        );
+      }
+      if (transactionRaw.gasPrice) {
+        transaction.gasPrice = BigNumber(
+          BigNumber(transactionRaw.gasPrice)
+            .multipliedBy(1.3)
+            .toFixed(),
+        );
+      }
+    }
```

Comment:
As mentionned before this will not work as all contract interactions have a 0 amount.

Also it feels really clunky that you have to "guess" at this stage if it's a cancel or speed up, so the `editType` Stepper prop could be useful here, even though I think at summary step, you should already have a finalized transaction, so every update should be done before (at method selection for me)

Reply:
fixed